modules/contracts/apps/ping-pong/tests/ping_pong_tests.rs

```diff
@@ -0,0 +1,317 @@
+    let mut i = chain.block_info().unwrap();
+    if i.height % 2 == 1 {
+        ()
+    } else {
+        i.height += 1;
+        chain.app.borrow_mut().set_block(i);
+    }
+}
+
+pub fn logger_test_init() {
+    let _ = env_logger::builder().is_test(true).try_init();
+}
+
+#[test]
+fn successful_install() -> anyhow::Result<()> {
+    logger_test_init();
+
+    // Create a sender and mock env
+    let mock_interchain =
+        MockBech32InterchainEnv::new(vec![(JUNO, "juno"), (STARGAZE, "stargaze")]);
+    let env = PingPong::setup(&mock_interchain)?;
+    let app1 = env.app;
+
+    let mock_stargaze = env.abs_stargaze.environment();
+
+    let wins = app1.wins()?;
+    assert_eq!(wins, WinsResponse { wins: 0, losses: 0 });
+
+    let module_addrs = env
+        .remote_account
+        .module_addresses(vec![APP_ID.to_owned()])?;
+    let wins: WinsResponse = mock_stargaze.query(
+        &ping_pong::msg::QueryMsg::from(AppQueryMsg::Wins {}),
+        &module_addrs.modules[0].1,
+    )?;
+    assert_eq!(wins, WinsResponse { wins: 0, losses: 0 });
+
+    let app2 = env.remote_account.application::<AppInterface<_>>()?;
+
+    let wins: WinsResponse = app2.wins()?;
+
+    assert_eq!(wins, WinsResponse { wins: 0, losses: 0 });
+    Ok(())
+}
+
+#[test]
+fn successful_ping_pong() -> anyhow::Result<()> {
+    logger_test_init();
+
+    // Create a sender and mock env
+    let mock_interchain =
+        MockBech32InterchainEnv::new(vec![(JUNO, "juno"), (STARGAZE, "stargaze")]);
+    let env = PingPong::setup(&mock_interchain)?;
+    let app = env.app;
+    let remote_app = env.remote_account.application::<AppInterface<_>>()?;
+
+    // Ensure account created
+    env.abs_stargaze
+        .version_control()
+        .account_base(AccountId::new(
+            1,
+            AccountTrace::Remote(vec![ChainName::from_chain_id(JUNO)]),
+        )?)?;
+
+    let wins = app.wins()?;
+    assert_eq!(wins, WinsResponse { losses: 0, wins: 0 });
+    let wins = remote_app.wins()?;
+    assert_eq!(wins, WinsResponse { losses: 0, wins: 0 });
+
+    // let stargaze win
+    set_to_win(mock_interchain.chain(STARGAZE)?);
+    set_to_lose(mock_interchain.chain(JUNO)?);
+
+    // juno plays against stargaze
+    let pp = app.ping_pong(ChainName::from_chain_id(STARGAZE))?;
+    mock_interchain.check_ibc(JUNO, pp)?.into_result()?;
+
+    // stargaze wins, juno lost.
+    let wins = app.wins()?;
+    assert_eq!(wins, WinsResponse { losses: 1, wins: 0 });
+
+    // now let juno win
+    set_to_lose(mock_interchain.chain(STARGAZE)?);
+    set_to_win(mock_interchain.chain(JUNO)?);
+
+    let pp = app.ping_pong(ChainName::from_chain_id(STARGAZE))?;
+    mock_interchain.check_ibc(JUNO, pp)?.into_result()?;
+
+    let wins = app.wins()?;
+    assert_eq!(wins.wins, 1);
+    assert_eq!(wins.losses, 1);
+
+    let wins: WinsResponse = remote_app.wins()?;
+    assert_eq!(wins.losses, 1);
+    Ok(())
+}
+
+// #[test]
+// fn successful_remote_ping_pong() -> anyhow::Result<()> {
```

Comment:
Let's add a test where pingpong executed on remote pingpong. Would be great way to test if `into_dest_account_id` work as intended